[stable/grafana] Allow secrets for sidecar (#13492)

fr-ser opened the pull request:
Updated the sidecar image to version 0.0.18
This allows also using secrets for volume mounts
This way the dashboard and data source import can be extended to those
datatypes

Signed-off-by: Sergej Herbert <herbert.sergej@gmail.com>


#### What this PR does / why we need it:
I pushed a PR into a used sidecar image (https://github.com/kiwigrid/k8s-sidecar) now we can use secrets for data source and dashboard provisioning for grafana.


#### Which issue this PR fixes
  - fixes #13492

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ x] Chart Version bumped
- [ x] Variables are documented in the README.md
- [ x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

fr-ser committed:
[stable/grafana] Allow secrets for sidecar (#13492)

Updated the sidecar image to version 0.0.18
This allows also using secrets for volume mounts
This way the dashboard and datasource import can be extended to those
datatypes

Signed-off-by: Sergej Herbert <herbert.sergej@gmail.com>

max-baz commented on stable/grafana/templates/clusterrole.yaml:
```diff
@@ -15,7 +15,13 @@ metadata:
 rules:
 - apiGroups: [""] # "" indicates the core API group
+  {{- if eq .Values.sidecar.resourceType "both" }}
+  resources: ["configmaps", "secrets"]
```
I understand you want to preserve backward compatibility and still support configmaps, but why even make it configurable instead of just hardcoding the "both" as the selected behavior?

fr-ser replied:
Good point. I wanted to use as little permissions as necessary. But we can go for both here, if that is not a "security concern"

max-baz commented on stable/grafana/README.md:
```diff
@@ -176,7 +177,7 @@ the url value is https://yourgerritserver/a/user%2Frepo/branches/master/files/di
 ## Sidecar for dashboards
 
-If the parameter `sidecar.dashboards.enabled` is set, a sidecar container is deployed in the grafana pod. This container watches all config maps in the cluster and filters out the ones with a label as defined in `sidecar.dashboards.label`. The files defined in those configmaps are written to a folder and accessed by grafana. Changes to the configmaps are monitored and the imported dashboards are  ...
+If the parameter `sidecar.dashboards.enabled` is set, a sidecar container is deployed in the grafana pod. This container watches all configmaps (or secrets) in the cluster and filters out the ones with a label as defined in `sidecar.dashboards.label`. The files defined in those configmaps are written to a folder and accessed by grafana. Changes to the configmaps are monitored and the imported das ...
```
To highlight that secrets are preferable, I propose replacing "watches all configmaps (or secrets)" with "watches all secrets (or configmaps, though not recommended)", plus adding a small note in a separate paragraph explaining _why_ secrets are a better choice.

fr-ser replied:
In order to stay backwards compatible I would have kept configmaps the default in the description.
But I can add a paragraph why secrets would be better. Is that a good common ground?

maorfr commented on stable/grafana/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
-version: 3.5.8
-appVersion: 6.2.4
+version: 3.5.9
+appVersion: 6.2.5
```
this represents the `grafana` image tag, so no need to update this if you are not updating the image.

fr-ser replied:
finally I know what this is :laughing:

maorfr commented on stable/grafana/README.md:
```diff
@@ -92,14 +92,15 @@ The command removes all the Kubernetes components associated with the chart and
 | `sidecar.dashboards.defaultFolderName`                | The default folder name, it will create a subfolder under the `sidecar.dashboards.folder` and put dashboards in there instead | `nil`                                |
 | `sidecar.dashboards.searchNamespace`      | If specified, the sidecar will search for dashboard config-maps inside this namespace. Otherwise the namespace in which the sidecar is running will be used. It's also possible to specify ALL to search in all namespaces | `nil`                                |
-| `sidecar.datasources.enabled`             | Enabled the cluster wide search for datasources and adds/updates/deletes them in grafana |`false`       |
+| `sidecar.datasources.enabled`             | Enables the cluster wide search for datasources and adds/updates/deletes them in grafana |`false`       |
```
thanks!

maorfr commented on stable/grafana/templates/clusterrole.yaml:
```diff
@@ -15,7 +15,13 @@ metadata:
+  {{- if eq .Values.sidecar.resourceType "both" }}
+  resources: ["configmaps", "secrets"]
+  {{- else if eq .Values.sidecar.resourceType "secret" }}
+  resources: [secrets"]
```
missing `"`

maorfr commented on stable/grafana/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
 name: grafana
-version: 3.5.8
-appVersion: 6.2.4
+version: 3.5.9
```
lets do a minor version bump

fr-ser committed:
[stable/grafana] Allow secret - review (#13492)

- keep the appVersion the same
- bump minor version
- add missing apostrophe

Signed-off-by: Sergej Herbert <herbert.sergej@gmail.com>

fr-ser committed:
[stable/grafana] Allow secret - review #2 (#13492)

- enable sidecar resource secrets and configmaps by default
- update readme to indicate secret preference for datasources

Signed-off-by: Sergej Herbert <herbert.sergej@gmail.com>

maorfr committed:
Merge branch 'master' into ISSUE#13492_grafana_datasources_as_secrets
